### PR TITLE
feat: Added Release Notes & Top Bar

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description
This PR closes ticket no :-
#972 
#973 

In this Pr i have  added two features :- 
1.Added  the Update Release Section in Help page in which data is from GitHub releases notes API .
2. Added a top bar which render conditionally if there is update available.

sparrow falds

sparrowafdf

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

File: src/packages/@app/pages/Dashboard/Dashboard.svelte [hunk 1]
```diff
@@ -63,6 +69,60 @@
...
+      console.error(error);
+    }
+  });
+
+  const handleUpdatePopUp = (flag: boolean) => {
+    updateAvailable = flag;
+  };
+
+  const initiateUpdate = async () => {
+    console.log("Inside inital update");
```
Comment (Astitva877) [inline src/packages/@app/pages/Dashboard/Dashboard.svelte:104]: remove this console

File: src/packages/@app/pages/Help/HelpPage.svelte [hunk 2]
```diff
@@ -1,18 +1,70 @@
...
+  }
+
+  let releaseNotesData = [];
+
+  onMount(async () => {
+    await _viewModel.fetchReleaseNotes();
+    releaseNotesData = await _viewModel.getAllReleaseNotes();
+    let pathname = window.location.pathname;
+    if (pathname.includes("app/help/updates")) {
+      activeTab = "updates";
```
Comment (Astitva877) [inline src/packages/@app/pages/Help/HelpPage.svelte:33]: Can we take this from enum or contants?

File: src/packages/@app/pages/Help/HelpPage.svelte [hunk 3]
```diff
@@ -1,18 +1,70 @@
...
+    } else {
+      activeTab = "feedback";
```
Comment (Astitva877) [inline src/packages/@app/pages/Help/HelpPage.svelte:35]: here also

File: src/packages/@app/pages/Dashboard/Dashboard.svelte [hunk 4]
```diff
@@ -63,6 +69,60 @@
...
+          await relaunch();
+        }, WAIT_TIME_BEFORE_RESTART_IN_SECONDS * 1000);
+      }
+      console.log("Done with  inital update");
+    } catch (e) {
+      notifications.error("Update Failed!");
+      console.error(e);
+    } finally {
+      showProgressBar = false;
+      updateAvailable = false;
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/Dashboard/Dashboard.svelte:123]: push business logic to dashboard view model

File: src/packages/@app/pages/Dashboard/Dashboard.svelte [hunk 5]
```diff
@@ -77,6 +137,13 @@
     {currentWorkspaceId}
     {currentWorkspaceName}
   />
+  {#if showProgressBar === true}
+  <ProgressBar onClick={handleUpdatePopUp} title="Update in progress" />{/if}
+  <Ticker
+    show={tickerVisible && updateAvailable}
+    {hideTicker}
+    onUpdate={initiateUpdate}
+  />
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/Dashboard/Dashboard.svelte:146]: onUpdate={viewmodel.initiateUpdate}

File: src/packages/@common/components/ticker/Ticker.svelte
Comment (itsmdasifraza) [inline src/packages/@common/components/ticker/Ticker.svelte:1]: Can we rename this file to "updater" or something descriptive

File: src/packages/@library/icons/CrossIconWhite.svelte [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<svg width="12" height="16" viewBox="0 0 12 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M10.7062 4.70469C11.0968 4.31406 11.0968 3.67969 10.7062 3.28906C10.3155 2.89844 9.68115 2.89844 9.29053 3.28906L5.9999 6.58281L2.70615 3.29219C2.31553 2.90156 1.68115 2.90156 1.29053 3.29219C0.899902 3.68281 0.899902 4.31719 1.29053 4.70781L4.58428 7.99844L1.29365 11.2922C0.903027 11.6828 0.903027 12.3172 1.29365 12.7078C1.68428 13.0984 2.31865 13.0984 2.70928 12.7078L5.9999 9.41406L9.29365 12.7047C9.68428 13.0953 10.3187 13.0953 10.7093 12.7047C11.0999 12.3141 11.0999 11.6797 10.7093 11.2891L7.41553 7.99844L10.7062 4.70469Z" fill="#F5F5F5"/>
+</svg>
```
Comment (itsmdasifraza) [inline src/packages/@library/icons/CrossIconWhite.svelte:3]: use props for height width and color, "24" "24" "grey"

Comment (itsmdasifraza) [reply src/packages/@library/icons/CrossIconWhite.svelte:3]: Remove "white" from file name

File: src/packages/@app/repositories/release.repository.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      .rxdb.releaseupdate.findOne({
+        selector: query,
+      })
+      .exec();
+
+    if (document) {
+      const res = await document.incrementalPatch(data);
+      return res;
+    } else {
+      throw new Error("Document not found");
```
Comment (itsmdasifraza) [inline src/packages/@app/repositories/release.repository.ts:61]: use console.error here, "throw" terminates the application if we dont use try catch

File: src/packages/@app/repositories/release.repository.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  /**
+   * @description - inserts document to the database
+   * @param data - document to be inserted
+   * @returns - inserted data
+   */
+  //   public insert = async (data: ReleaseDocType): Promise<ReleaseDocType> => {
+  //     const document = await RxDB.getInstance().rxdb.release.insert(data);
+  //     return document;
+  //   };
+
```
Comment (itsmdasifraza) [inline src/packages/@app/repositories/release.repository.ts:38]: ??

File: src/packages/@support/common/constants/learnMore.constant.ts
Comment (itsmdasifraza) [inline src/packages/@support/common/constants/learnMore.constant.ts:1]: create index.ts in constant folder and export learnmore.constant from there, take  learn more value from env

Comment (Abhishek9503) [reply src/packages/@support/common/constants/learnMore.constant.ts:1]: The link will be constant throughout  later  and it is a public link  so does it make sense to put it in the env file .?

**Review by Astitva877 — APPROVED** [review]

**Review by Astitva877 — APPROVED** [review]

**Review by gc-codes — APPROVED** [review]

**Comment by shreyas070707** [comment]
currently not able to test
cc- @ayushadhikari93  @sanilnayak

**Pull request merged**